client/src/views/list-order/ListOrder.js

```diff
@@ -2,20 +2,21 @@ import React, { useState, useEffect } from 'react';
 import axios from 'axios';
 import './listOrder.scss';
 
-function ListOrder() {
+function ListOrder(props) {
   // Declare a new state variable, which we'll call "count"
   const [orderList, setOrderList] = useState();
 
   useEffect(() => {
     // Fetch the data on load
-    axios.get('http://localhost:3001/listorder', { withCredentials: true }).then((response) => {
+    const parameters = props.customerId? ('?customerId=' + props.customerId) : '';
+    axios.get('http://localhost:3001/listorder' + parameters, { withCredentials: true }).then((response) => {
       setOrderList(response.data);
     });
-  }, []);
+  }, [props.customerId]);
 
   return (
     <div>
-      <h1>Orders</h1>
+      <h1>{props.customerId && "Your "}Orders</h1>
```

Comment:
I'm a bit confused about this line. If you don't find a valid customer id, we just write "Orders"? How is this case even possible, when we force login validation for this page?

Reply:
We don't force login validation for this page. In fact, not having a customer ID is the default behaviour for the ListOrder page. It is only as of this PR that the ListOrder page is even aware of what a customerId is. What I did was reuse the ListOrder component that was previously used to list all orders, and pass it a customerId so that when *I* use it in my page, it only fetched the orders for the given customer. 

Thus, the above conditional ensures that we don't see "Your Orders" if we click on List Orders, but rather simply "Orders", as was the case all along.

Reply:
Thanks for the clarification. That now makes more sense. In this case, I have different feedback: rather than re-using the component in this way, it would make sense to have a generic "OrderList" component which only understands how to render a given list of orders in a table. Then, each of these pages could have an OrderList component, each being passed their respective data.

This is, of course, not very important since the current way works, but generally speaking the golden rule of components is that they are self-contained and encapsulated. In this case, the component has two different functionalities based on its parent. It's certainly not a big deal, but something to consider for future component designs.